Milestone 1: MozaicDot

sushmitsarmah opened the pull request:
# 📬 Milestone Delivery Checklist

- [x] The [delivery template](https://github.com/Polkadot-Fast-Grants/delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] The milestone is complete (no partial deliveries).
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] The delivery is according to the [delivery guidelines](https://github.com/Polkadot-Fast-Grants/delivery/blob/master/delivery-guidelines.md).
- [x] I understand that the milestone will be evaluated within 14 days, and if changes are needed, I will have one opportunity to fix and resubmit within 14 days.

Link to the application pull request: https://github.com/Polkadot-Fast-Grants/apply/pull/1

sushmitsarmah commented:
Hi @sacha-l 

1. The repository [MozaicDot](https://github.com/sushmitsarmah/mozaic_dot_nfts) is public.
2. The web app defaults to Paseo AssetHub now
3. Deliverables 0a-d has been added. 0c is the user testing guide.
4. You can login with your wallet like sub wallet. with some PAS in your wallet you can create a collection and NFT in it, view it, edit it, delete it, set a price to sell and list it, login with another wallet and buy that NFT. I have tested everything on the website.

FFilipUnique reviewed (changes requested):
Hi @sushmitsarmah. Reviewed the code. Well structured. Tested the collection creation, nft mint and I have verified events on-chain. I have just a few comments regarding the codebase you might want to take into consideration:
## 0a - 0d
0. You provided the following points of delivery in your proposal:

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| 0a. | License | Apache 2.0 |
| 0b. | Documentation | Full inline documentation and user tutorial |
| 0c. | Testing and Testing Guide | Comprehensive unit and integration tests |
| 0d. | Article | Publication on AI-driven NFTs in Polkadot |

The documentation and guides could have been delivered in the demo application, I see the page links are in place but they do not lead to the docs. It would be appropriate to fulfill 0b. At least a guide on the what is available and what can be tested.
The 0c would have been helpful to us so we can focus on specific steps to perform to confirm and verify intended operation.
And 0d would have been just fine, we are interested in how you approach this issue.

## app network switch error
Switching to Unique Network throws an error
Error
Failed to fetch collections: Server responded with Error 404 - : {"message":"Route GET:/pallet-nfts/account/collections?account=5H9EfoFvYKAh2qAYmtxQ8GL73P1xokJidPL2sveZVnC6FhAU not found","error":"Not Found","statusCode":404}

ping the Dev Support tg channel to help you fix that.

## code review notes
1. in UniqueSDKProvider, a switch network fail would fail silently. User would never know it failed until later.
2. In the Create.tsx section you might want to consider only Unique network main chain of the three chain.
related to Unique Network. Quartz is closing public access and Opal will be shut down soon.
3. Catch 'Zero funds' events gracefully, notify user so they know they don't have funds in the wallet for the network they chose.

Appart from the 0a - 0d not being delivered, all else is not an issue.
Please provide these expected delivery points. Otherwise all else is good for approval from my point of view.

Ahhhh, I followed a stale link. I see you've updated 0a - 0d.  All clear then 👍

sushmitsarmah commented:
hi @FFilipUnique @sacha-l 

1. Fixed the a silent switch network failure. Now user will be notified and network won't change in the UI if network change failure.
2. Removed Quartz, Opal, Westend and Rococo networks.
3. if zero funds then user gets a notification and it is also displayed in the navigation bar so user always knows if they don't have funds.
4. As for the Unique Network switch failure, the issue was because of collections from other networks. now all collections are cleared when a network is switched. as for the actual 404 error I have messaged the dev team on telegram.
5. 0a - 0d are all updated as you already noticed @FFilipUnique 

Thanks
